modules/apps/adaptive-media/adaptive-media-image-impl/src/main/java/com/liferay/adaptive/media/image/internal/util/comparator/AMAttributeDistanceComparator.java

```diff
@@ -65,14 +65,18 @@ public long compare(
 			AMAttribute<AMImageProcessor, Object> amAttribute =
 				sortCriterion.getKey();
 
-			Optional<?> valueOptional1 = adaptiveMedia1.getValueOptional(
-				amAttribute);
-			Optional<?> valueOptional2 = adaptiveMedia2.getValueOptional(
-				amAttribute);
-
-			Optional<Long> valueOptional3 = valueOptional1.flatMap(
-				value1 -> valueOptional2.map(
-					value2 -> amAttribute.compare(value1, value2)));
+			Object value1 = adaptiveMedia1.getValue(amAttribute);
+			Object value2 = adaptiveMedia2.getValue(amAttribute);
+
+			Optional<Long> valueOptional3;
```

Comment:
@slnn Let's clean this up as long as it's method internal and related to the logic you modified.